### PR TITLE
Add "Clear missing files" option to recent ISO list

### PR DESCRIPTION
**Pull request by italodirenzo93**
### Description of Changes
Adds an option to the recent ISO list to remove all entries that can no longer be found in their original location.

### Rationale behind Changes
Solution for #4963. I decided on using the term "prune" instead of "clean" because it sounds too similar to the existing "clear". Would welcome input on what the terminology for this function should be.

### Suggested Testing Steps
1. Load an ISO file into PCSX2
2. In a separate file browser window, move that file to a different location on disk
3. Click the "Prune ISO list" option and the file will be removed from the list

### CODE REVIEW

**Comment by lightningterror** [comment]
Maybe `Clear invalid ISO list` would make more sense.

**Comment by IlDucci** [comment]
> 
> 
> Maybe `Clear invalid ISO list` would make more sense.

Or simply "Clear invalid ISO files". Maybe things could be made more obvious by referring to the entire ISO list (for the full removal) and to the individual ISO files (for the pruning of only any missing ISO files).

**Comment by refractionpcsx2** [comment]
Or an even more user friendly way of writing it
`Clear missing files`

**Comment by nf6429** [comment]
Agree with the above. "Clear invalid ISO" to an end user could indicate something is wrong with your file itself, not that it's just been moved, whilst "clear missing files" directly states what the issue is.

**Comment by IlDucci** [comment]
> 
> 
> Agree with the above. "Clear invalid ISO" to an end user could indicate something is wrong with your file itself, not that it's just been moved, whilst "clear missing files" directly states what the issue is.

Agreed as well.

**Comment by arcum42** [comment]
My main suggestion would be to show a list of what's going to be removed on the confirmation dialog box, before you actually remove it. Otherwise, I could see times where you could accidentally remove all or most of the entries because of something not being mounted at the moment...

**Comment by RedDevilus** [comment]
```
* Game1
* Game2
* Game3
These will be removed from the listing. Do you confirm?
OK              Cancel
```

-----
Should be feasible to do something like this.

**Comment by darkxex** [comment]
I find it very useful.

**Comment by italodirenzo93** [comment]
I agree. Will update the PR with a new commit soon.

**Comment by RedDevilus** [comment]
On your location you can't fully avoid WX but if you can, sure.

File: pcsx2/gui/MainMenuClicks.cpp [hunk 1]
```diff
@@ -583,6 +583,46 @@ void MainEmuFrame::Menu_IsoClear_Click(wxCommandEvent& event)
...
+		for (const auto& file : missing_files)
+		{
+			missing_files_list += "* ";
+			missing_files_list += file.Filename;
+			missing_files_list += "\n";
+		}
+
+		dialog += dialog.Text(missing_files_list);
+
+		bool confirmed = pxIssueConfirmation(dialog, MsgButtons().YesNo()) == wxID_YES;
```
Comment (lightningterror) [inline pcsx2/gui/MainMenuClicks.cpp:613]: Const.

**Review by lightningterror — COMMENTED** [review]
Commits should be squashed and properly named with prefixes.

**Review by lightningterror — APPROVED** [review]
Needs to be tested.

**Comment by arcum42** [comment]
Works properly, from what I can see.

**Review by arcum42 — APPROVED** [review]

**Pull request merged**